packages/flutter/lib/src/material/pickers/input_date_picker.dart

```diff
@@ -227,54 +223,58 @@ class _InputDatePickerFormFieldState extends State<InputDatePickerFormField> {
     return OrientationBuilder(builder: (BuildContext context, Orientation orientation) {
       assert(orientation != null);
 
-      return Container(
-        padding: const EdgeInsets.symmetric(horizontal: 24),
-        height: orientation == Orientation.portrait ? _inputPortraitHeight : _inputLandscapeHeight,
-        child: Column(
-          children: <Widget>[
-            const Spacer(),
-            TextFormField(
-              decoration: InputDecoration(
-                border: const UnderlineInputBorder(),
-                filled: true,
-                // TODO(darrenaustin): localize 'mm/dd/yyyy' and 'Enter Date'
-                hintText: widget.fieldHintText ?? 'mm/dd/yyyy',
-                labelText: widget.fieldLabelText ?? 'Enter Date',
-              ),
-              validator: _validateDate,
-              inputFormatters: <TextInputFormatter>[
-                // TODO(darrenaustin): localize date separator '/'
-                _DateTextInputFormatter('/'),
-              ],
-              keyboardType: TextInputType.datetime,
-              onSaved: _handleSaved,
-              onFieldSubmitted: _handleSubmitted,
-              autofocus: widget.autofocus,
-              controller: _controller,
-            ),
-            const Spacer(),
-          ],
+      return TextFormField(
+        decoration: InputDecoration(
+          border: const UnderlineInputBorder(),
+          filled: true,
+          // TODO(darrenaustin): localize 'mm/dd/yyyy' and 'Enter Date'
+          hintText: widget.fieldHintText ?? 'mm/dd/yyyy',
+          labelText: widget.fieldLabelText ?? 'Enter Date',
         ),
+        validator: _validateDate,
+        inputFormatters: <TextInputFormatter>[
+          // TODO(darrenaustin): localize date separator '/'
+          DateTextInputFormatter('/'),
+        ],
+        keyboardType: TextInputType.datetime,
+        onSaved: _handleSaved,
+        onFieldSubmitted: _handleSubmitted,
+        autofocus: widget.autofocus,
+        controller: _controller,
       );
     });
   }
 }
 
-class _DateTextInputFormatter extends TextInputFormatter {
-
-  _DateTextInputFormatter(this.separator);
-
+/// A `TextInputFormatter` set up to format dates.
+///
+/// Note: this is not publicly exported (see pickers.dart), as it is
```

Comment:
Why not private?

Reply:
Because it is shared between the `InputDatePickerFormField` and the `InputDateRangePicker` which is in another file. I could move the range picker into the `input_date_picker.dart` file, but `InputDateRangePicker` is not a public API either.

Again a package private feature would be welcome. I didn't see many examples of how to handle this sort of situation elsewhere in the framework. Most components are as complicated as these date pickers, so they just put everything in one large files. With the date pickers that just seemed too unwieldy.